Public/Src/Tools/BuildXL.Tools.CredentialProvider/CredentialProviderGenerator.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace BuildXL.Tools.CredentialProvider
+{
+    class CredentialProviderGenerator
+    {
+        private static Dictionary<string, string> ValidPackageSources = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+            { "https://pkgs.dev.azure.com/1essharedassets/_packaging/BuildXL/nuget/v3/index.json", "1ESSHAREDASSETS_BUILDXL_FEED_PAT" },
+            { "https://pkgs.dev.azure.com/cloudbuild/_packaging/BuildXL.Selfhost/nuget/v3/index.json", "CLOUDBUILD_BUILDXL_SELFHOST_FEED_PAT" }
+        };
+
+        public static int Main(string[] args)
+        {
+            string uri = GetUri(args);
+
+            if (string.IsNullOrWhiteSpace(uri))
+            {
+                return 2;

```

Comment:
In other error cases you also emit an error message, so it would make sense to do the same here as well.